### ISSUE TITLE
react-split is not happy if the children change or the elements are re-generated

### ISSUE THREAD

**Issue by greggman** [nathancahill/split#390]
As a simple example render N children where N changes over time

My impression is because react-split is using split and split is manually manipulating the DOM, then if react comes through and re-arranges the DOM (like if it deletes the react based children and generates new ones) then now Split is looking at the old removed DOM elements among other issues.

This isn't my actual issue but it was the smallest thing I could think of

https://jsfiddle.net/greggman/dqwe06cn/

You'll see we start with 3 children and there are 2 splitters. Then one child is removed (not rendered in react) and we see there are still 2 splitters, one above the top child. Then we add the child back (rendered in react) and now there's a splitter, 2 elements, a splitter and an element.

To put more visually it starts like this

     pane splitter pane splitter pane

then becomes

     splitter pane splitter pane

then

     splitter pane pane splitter pane

This is because react-split it's not actually a react component, it's more of a non-react library kind of patched into kind of sometimes work in react?

In my actual app I'm not doing anything like removing children, all that's happening is react is re-rendering (re-creating) DOM elements and Split ends up confused.

**Comment by jacobjzhang** [comment]
Yeah, I've noticed this too. I have a nested split setup like the following:

```
   A  |
 -----|  C
   B  |
```

I tried have the option to close panel `B` if desired, to result in:

```
       |
   A   |  C
       |
```

But what ends up happening is like the above description -- it looks like things don't get cleaned up properly/re-rendered correctly. The splitter between `A` and `B` doesn't disappear, and `A` takes the width of the entire page (rather than the expected 50%). Any ideas on how to get around this?

**Comment by greggman** [comment]
I ended up writing my own https://www.npmjs.com/package/react-split-it

**Comment by nathancahill** [comment]
I'd like to improve how Split handles React children changing. Open to ideas!
